File: HahnCatFacts.API/HahnCatFacts.Infrastructure/Persistence/Repositories/CatFactRepository.cs
using HahnCatFacts.Domain.Entities;
using HahnCatFacts.Domain.Interfaces;
using HahnCatFacts.Infrastructure.Persistence.DataBaseContext;
using Microsoft.EntityFrameworkCore;

namespace HahnCatFacts.Infrastructure.Persistence.Repositories
{
    public class CatFactRepository : ICatFactRepository
    {
        private readonly ApplicationDbContext _dbContex;

        public CatFactRepository(ApplicationDbContext context)
        {
            _dbContex = context;
        }

        public IQueryable<CatFact> GetAllNoTracking()
        {
            return _dbContex.CatFacts.AsNoTracking();
        }

        public async Task AddAsync(CatFact entity)
        {
            await _dbContex.CatFacts.AddAsync(entity);
            await _dbContex.SaveChangesAsync();
        }
    }
}

File: HahnCatFacts.UI/src/data/repositories/CatFactRepository.ts
import { CatFact } from "@/domain/entities/CatFact";
import { ICatFactRepository } from "@/domain/repositories/ICatFactRepository";
import axios from "axios";

const apiUrl = process.env.VUE_APP_API_URL + "/CatFact";

export class CatFactRepository implements ICatFactRepository {
    async getAllCatFacts(): Promise<CatFact[]> {
        const response = await axios.get<CatFact[]>(apiUrl); 
        
        return response.data;
    }
}